### PR TITLE
feat: add in a job to automatically generate and sync daily

### PR DESCRIPTION
**Pull request by ckipp01**
I was about to sync and send in a pr again, but I figured why not just automate this. This change adds a new CI job that will run every morning and detect if there are changes. If so, it will commit them and send in a pr.

### CODE REVIEW

File: .github/workflows/sync.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          files: |
+            src/grammar.json
+            src/node-types.json
+            src/parser.c
+            src/tree_sitter/parser.h
+
+      - name: Commit changes if necessary
+        if: steps.check-for-changes.outputs.files_changed == 'true'
+        run: |
+          git config user.name "GitHub"
```
Comment (keynmol) [inline .github/workflows/sync.yml:37]: So github just provides an account you can use to do things like these?
Wild if so - I'm tempted to just merge this and run it manually to see if it works!

File: .github/workflows/sync.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          LAST_COMMIT=$(git log --format-"%H" -n 1)
+          git commit -m "chore: generate and sync from $UPDATED_PARSERS"
+          git clean -xf
+
+      - name: Create Pull Request
+        if: steps.check-for-changes.outputs.files_changed == 'true'
+        uses: peter-evans/create-pull-request@v4
+        with:
+          title: "chore: generate and sync latest changes"
+          branch: generation
```
Comment (keynmol) [inline .github/workflows/sync.yml:49]: where are you `git checkout -b` this branch? or does this GHA do it for you?

File: .github/workflows/sync.yml (hunk 1 above)
Comment (ckipp01) [reply .github/workflows/sync.yml:37]: Seems so, I took a lot of this from the way nvim-treesitter does this for their lock file.

File: .github/workflows/sync.yml (hunk 2 above)
Comment (ckipp01) [reply .github/workflows/sync.yml:49]: So I think technically we just commit it to master, but then the next step in `create-pull-request` specifies the `branch`. I think it must just take those changes, create a new branch, and then create the pr. Sort of copying this part from https://github.com/nvim-treesitter/nvim-treesitter/blob/5e9c6b6ba079a90a2545e651e3740b655b9ff035/.github/workflows/update-lockfile.yml#L40-L56.

**Review by eed3si9n — APPROVED** [review]
Thanks @ckipp01!

**Comment by ckipp01** [comment]
Alright, I'm gonna merge and we'll try it out. I fully expect _something_ to not work right, but we'll see 😄

**Pull request merged**